crates/orchestrator/src/tests/database/mod.rs

```
@@ -158,50 +161,42 @@ async fn database_get_jobs_after_internal_id_by_job_type_works() {
-
-    assert!(updating_job_res, "Job result assertion failed");
-}
-
-/// Test for `update_job_status` operation in database trait.
-/// Failing Case : Creating a job with version 1 and updating the job with version 0 update only.
-#[rstest]
-#[tokio::test]
-async fn database_update_job_status_failing_case_works() {
-    let services = TestConfigBuilder::new().configure_database(ConfigType::Actual).build().await;
-    let config = services.config;
-    let database_client = config.database();
-
-    // Scenario :
-
-    // Worker 1 :
-    // Job is created
-    let job = build_job_item(JobType::SnosRun, JobStatus::Created, 1);
-    database_client.create_job(job.clone()).await.unwrap();
-
-    // Worker 2 :
-    // Job is updated
-    let updating_job_res = database_client.update_job_status(&job, JobStatus::Completed).await.is_ok();
-    assert!(updating_job_res, "Job result assertion failed");
-
-    // Worker 1 :
-    // Job update try (update should fail)
-    let updating_job_res = database_client.update_job_status(&job, JobStatus::PendingVerification).await.is_err();
-    assert!(updating_job_res, "Job result assertion failed");
+    let job_id = job.id;
+
+    let metadata = HashMap::new();
+    let key = "test_key";
+    let updated_metadata = increment_key_in_metadata(&metadata, key).unwrap();
+
+    let job_cloned = job.clone();
+    let _ = database_client
+        .update_job(&job_cloned, JobItemUpdates {
+            status: Some(JobStatus::LockedForProcessing),
+            metadata: Some(updated_metadata),
+            external_id: None,
+            internal_id: None,
+            job_type: None,
+        })
+        .await;
+
+    let mut job_after_updates = build_job_item(JobType::DataSubmission, JobStatus::LockedForProcessing, 456);
+    job_after_updates.version += 1;
+
+    if let Some(job_after_updates_db) = database_client.get_job_by_id(job_id).await.unwrap() {
```

we should throw an error if it's none?

---

Valid, added a [panic else case](https://github.com/madara-alliance/madara-orchestrator/pull/134/commits/e678708abeaa7175eed692b07d6c2c02fbe88f40).